unlock-app/src/services/walletService.js

```diff
@@ -91,23 +95,58 @@ export default class WalletService extends EventEmitter {
   }
 
   /**
-   * Function which yields the address of the account on the provider or creates a key pair.
+   * Poll to see if account has changed
    */
-  getAccount() {
-    const getOrCreateAccount = this.web3.eth.getAccounts().then(accounts => {
-      if (accounts.length === 0) {
-        return this._createAccount()
-      } else {
-        return Promise.resolve(accounts[0]) // defaults to the first account for now
-      }
-    })
+  async pollForAccountChange(isServer) {
+    if (isServer) return
+    try {
+      await delayPromise(POLLING_INTERVAL)
+    } catch (e) {
+      return // if the timeout throws, we don't start the loop, this is primarily for testing
+    }
+    try {
+      this.account = await this.checkForAccountChange(this.account)
+    } catch (e) {
+      // if the account poll fails, silently ignore it
+    }
+    // because this is async, the call stack is not affected, and does not grow
+    // for a non-async function this would quickly fill all available memory
+    this.pollForAccountChange()
+  }
+
+  /**
+   * given the prior account address, determine if the account has changed and return the
+   * current address regardless of any change
+   */
+  async checkForAccountChange(priorAddress) {
+    const nextAddress = await this._getAccountAddress(Promise.resolve(null))
```

Comment:
What is this promise passed as argument?

Reply:
Maybe name the function in a way that makes it more obvious what the argument is? `_getAccountAddressOrFallback`  for example ;)